### ISSUE TITLE
Shap Deep Explainer is giving irrelevant results

### ISSUE THREAD

**Issue by Shivayv** [shap/shap#1238]
Hello , 

I have a built keras model to predict a binary classification problem. Although while trying to use deepexplainer, i am unable to run it and then used `check_additivity = False`. This gave an output but was completely wrong. Am I missing something here ? 

```
#we use the first 1000 training examples as our background dataset to integrate over
explainer = shap.DeepExplainer(best_model, X_train[:1000].values)
shap_values = explainer.shap_values(X_test[:10].values, check_additivity = False)
# init the JS visualization code
shap.force_plot(explainer.expected_value[0], shap_values[0][0,:], X_test.iloc[0,:])

np.sum(shap_values[0][0,:])
```

![image](https://user-images.githubusercontent.com/8288263/83138396-58cf1c80-a108-11ea-96fb-90cac20dff87.png)

Given, it is a binary classification the final value has to be between 0 and 1 , but the values here are in 100's

**Comment by Shivayv** [comment]
Interestingly KernelSHAP does give relevant results :- why is it so ? 
expected value is correct and the magnitude of the shap values are irrelevant. 
Is this going to be looked into in TF2.0 ? 

![image](https://user-images.githubusercontent.com/8288263/83776276-9d792b80-a6a5-11ea-85e8-97d932bca354.png)

**Comment by ghost** [comment]
I am suprised that you can make DeepExplainer work at all. Quite a few people, including myself, have reported that they cannot make it work with TF2.x. Maybe I can try set check_additivity=False too. Do you know what it does?

What you showed here are probably the raw output scores. These are the scores before the output activiation function. Did you use Sigmoid?

**Comment by Shivayv** [comment]
@kdlin I am using tensorflow ==1.14.0. I know that `DeepExplainer` doesn't support TF2.X.  I did try applying a logit on the raw values ( sum of all shap values for a row) and found that values did not match the probability. Let me try again and get back on that. 

As far as `check_additivity`  - this checks if the output from shap computation equals the model output.

**Comment by ghost** [comment]
@Shivayv Do you use keras too?

**Comment by Shivayv** [comment]
@kdlin Yes I do. is that a problem ? or something related to the issue ?

**Comment by ghost** [comment]
@Shivayv I recalled that I had got an error message stating keras is not supported anymore, use tf.keras instead. However, to my knowledge, tf.keras is only supported in tf v2.x. It is confusing what is supported currently.

**Comment by shiva24yv** [comment]
@slundberg  @kdlin  I think i resolved this issue. The core reason that the shap values were irrelevant was due to the datatype of independent variables. Since I was one hot encoding all my independent variables - the data type was `int8` if you convert that to `int32/64`  the `DeepExplainer` works.

**Comment by Shivayv** [comment]
@slundberg @kdlin I think i resolved this issue. The core reason that the shap values were irrelevant was due to the datatype of independent variables. Since I was one hot encoding all my independent variables - the data type was int8 if you convert that to int32/64 the DeepExplainer works.

`for col in data.columns : 
      data[col] = data[col].astype(int)`

**Comment by ghost** [comment]
@Shivayv Thank you for the update. Glad it works for you now. Do you still have to use tensorflow ==1.14.0? Any idea when it will work with TF V2.x?
